presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -153,6 +153,15 @@ public long getSizeInMemory()
         return sizeInMemory;
     }
 
+    /**
+     * building hash sorted results requires memory for sorting group IDs.
+     * This method returns size of that memory requirement.
+     */
+    public long getGroupIdsSortingSize()
+    {
+        return getGroupCount() * Integer.BYTES;
```

Comment:
I don't think this is accurate. A `ArrayList<Integer>` uses more than 4 bytes per entry, because elements are `Object`s. Instead I would make this more explicit by pre-allocating the array used by `hashSortedGroupIds` and use `IntBigArray` then you can just use `sizeOf`